dev/sparktestsupport/modules.py

```diff
@@ -19,10 +19,67 @@
 import itertools
 import os
 import re
+import unittest
+import sys
+
+from sparktestsupport import SPARK_HOME
 
 all_modules = []
 
 
+def _get_module_from_name(name):
+    __import__(name)
+    return sys.modules[name]
+
+
+def _discover_python_unittests(*paths, discover_slow=False):
+    """Discover the python module which contains unittests under paths.
+
+    Such as:
+    ['pyspark/tests'], it will return the set of module name under the path of pyspark/tests, like
+    {'pyspark.tests.test_appsubmit', 'pyspark.tests.test_broadcast', ...}
+
+    Parameters
+    ----------
+    paths : str
+        Paths of modules to be discovered.
+    discover_slow : bool
+        If True, will only discover slow tests
+        If False, will discover all tests except slow tests
+
+    Returns
+    -------
+    A set of complete test module name discovered under specified paths
```

Comment:
It is possible to add a simple doctests at https://github.com/apache/spark/blob/master/dev/run-tests.py? The doctests there are ran before running the script.

Reply:
Yep

Reply:
https://github.com/apache/spark/pull/32867/commits/ef460a70716b3a4b4c9835b8d7fd73f839500e24#diff-baa74aac64952412da46580040412f216c4174a61a9f9bd201d41f0b84a50254R783-R793

Reply:
Emm, looks like it a diffcult to add this doctest, the doctest requires real import of pyspark, so it would be failed when running the testcase in run-test.py, but we still could run the local test at least. So, I will add the doctest in next commits.

Reply:
kk that's fine